mithril-stm/src/multi_sig.rs

```diff
@@ -269,12 +258,10 @@ impl ProofOfPossession {
     pub fn to_bytes(self) -> [u8; 96] {
         let mut pop_bytes = [0u8; 96];
         pop_bytes[..48].copy_from_slice(&self.k1.to_bytes());
-        let k2_bytes = unsafe {
-            let mut bytes = [0u8; 48];
-            blst_p1_compress(bytes.as_mut_ptr(), &self.k2);
-            bytes
-        };
-        pop_bytes[48..].copy_from_slice(&k2_bytes);
+
+        let k2_bytes = compress_p1(&self.k2);
+
+        pop_bytes[48..].copy_from_slice(k2_bytes.as_slice());
```

Comment:
It's probably readable enough to skip the k2_bytes variable (using `[..]` is the same as `as_slice()`:

```suggestion
        pop_bytes[48..].copy_from_slice(compress_p1(&self.k2)[..]);
```